bonfire/bonfire.py

```diff
@@ -793,9 +795,21 @@ def _get_apps_config(source, target_env, ref_env, local_config_path):
                 for component in app_cfg.get("components", []):
                     component["ref"] = "master"
 
-        # override any apps that were defined in 'apps' setion of local config file
-        apps_config.update(get_local_apps(config, fetch_remote=False))
+        # override any components that were defined in an apps components section of local config
+        # file
+        for app_name, app_cfg in apps_config.items():
+            if app_name not in local_apps.keys():
+                continue
+            for idx, component in enumerate(app_cfg.get("components", [])):
+                matched_components = list(
+                    filter(
+                        lambda c: component["name"] == c["name"], local_apps[app_name]["components"]
+                    )
+                )
 
+                # There can be only one
+                if len(matched_components) == 1:
```

Comment:
What if you want to add a new component in your local apps config that is not yet in app-interface? The match result would be `0` and I think in that case we could just append the local `component` to `app_cfg["components"]` right?

Should we raise an error if `len(matched_components) > 1` ?

Reply:
Well if it's not in app-interface, this code won't run right? That's covered by the if statement below? I felt like softly not doing anything instead of dying was the right course, but up to you if we want to raise.

Reply:
I am thinking of this scenario...

Local config looks like this:
```
apps:
  - name: ingress
    components:
    - name: some-new-component
```

and an app named `ingress` exists in app-interface and it has other components defined already.

What if 'some-new-component' does not exist yet in app-interface? Right now I think your logic would filter this component out. You probably still want 'some-new-component' to go in if you type `bonfire deploy ingress` right?

Reply:
Thinking on this further ... what if you have an entire 'app' in your local apps that is not yet in app-interface? The old logic would add it to your config. This new logic would omit it.

Reply:
I pushed some new logic that should address these cases